### PR TITLE
Replace gtk-doc with gi-docgen

### PR DESCRIPTION
**Pull request by tintou**
The gtk-doc project is dormant and most GLib-based libraries have now switched to gi-docgen.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -761,6 +761,23 @@ endif()
 
 libical_option(ICAL_BUILD_DOCS "Build documentation" True)
 if(ICAL_BUILD_DOCS)
+  libical_option(ICAL_GLIB_BUILD_DOCS "Build libical-glib documentation" True)
```
Comment (winterz) [inline CMakeLists.txt:764]: At first I was against making the default True.   but these days, especially with vcpkg support (Windows), I think this is fine.

But I think then we also need to make GOBJECT_INTROSPECTION try by default

**Review by winterz — CHANGES_REQUESTED** [review]
- I notice Install.txt still refers to `ENABLE_GTK_DOC`. please change to `DICAL_GLIB_BUILD_DOCS`

- In CMakeLists.txt you can remove

```
else()
  #to quiet a warning about unused manually-specified variables
  set(ENABLE_GTK_DOC False)
```
- and I think `GOBJECT_INTROSPECTION` needs to be true by default now

**Review by winterz — COMMENTED** [review]
also add a blurb about this in the ReleaseNotes.txt

File: CMakeLists.txt (hunk 1 above)
Comment (tintou) [reply CMakeLists.txt:761]: Made it default to True

**Comment by winterz** [comment]
testing this patch and get 2 failures:

```
[1176]$ ctest . -R libical-glib --rerun-failed --output-on-failure
Test project /home/allen/projects/libical/libical/build-testgcc1builtin-gcc
    Start 52: libical-glib-array
1/2 Test #52: libical-glib-array ...............***Failed    0.09 sec
Traceback (most recent call last):
  File "/home/allen/projects/libical/libical/src/test/libical-glib/array.py", line 27, in <module>
    ICalGLib.Timezone.array_append_from_vtimezone(array, zone0.get_component())
                                                         ^^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get_component'

    Start 54: libical-glib-component
2/2 Test #54: libical-glib-component ...........***Failed    0.10 sec
Traceback (most recent call last):
  File "/home/allen/projects/libical/libical/src/test/libical-glib/component.py", line 326, in <module>
    main()
  File "/home/allen/projects/libical/libical/src/test/libical-glib/component.py", line 308, in main
    ICalGLib.Timezone.set_tzid_prefix(zone.get_tzid().replace("Europe/Prague", ""))
                                      ^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'get_tzid'


0% tests passed, 2 tests failed out of 2

Total Test time (real) =   0.20 sec

The following tests FAILED:
         52 - libical-glib-array (Failed)
         54 - libical-glib-component (Failed)
```

**Comment by tintou** [comment]
> using the `-DUSE_BUILTIN_TZDATA=True` cmake option for testing this patch I get 2 failures:
> 
> ```
> [1176]$ ctest . -R libical-glib --rerun-failed --output-on-failure
> Test project /home/allen/projects/libical/libical/build-testgcc1builtin-gcc
>     Start 52: libical-glib-array
> 1/2 Test #52: libical-glib-array ...............***Failed    0.09 sec
> Traceback (most recent call last):
>   File "/home/allen/projects/libical/libical/src/test/libical-glib/array.py", line 27, in <module>
>     ICalGLib.Timezone.array_append_from_vtimezone(array, zone0.get_component())
>                                                          ^^^^^^^^^^^^^^^^^^^
> AttributeError: 'NoneType' object has no attribute 'get_component'
> 
>     Start 54: libical-glib-component
> 2/2 Test #54: libical-glib-component ...........***Failed    0.10 sec
> Traceback (most recent call last):
>   File "/home/allen/projects/libical/libical/src/test/libical-glib/component.py", line 326, in <module>
>     main()
>   File "/home/allen/projects/libical/libical/src/test/libical-glib/component.py", line 308, in main
>     ICalGLib.Timezone.set_tzid_prefix(zone.get_tzid().replace("Europe/Prague", ""))
>                                       ^^^^^^^^^^^^^
> AttributeError: 'NoneType' object has no attribute 'get_tzid'
> 
> 
> 0% tests passed, 2 tests failed out of 2
> 
> Total Test time (real) =   0.20 sec
> 
> The following tests FAILED:
>          52 - libical-glib-array (Failed)
>          54 - libical-glib-component (Failed)
> ```

I don't think that this is related to this MR (I don't see why it would be related :thinking: )

**Comment by winterz** [comment]
> I don't think that this is related to this MR (I don't see why it would be related 🤔 )
agree.  I'll look at fixing those in a separate patch.

**Review by winterz — APPROVED** [review]

**Pull request merged**